Support for docstrings in scenario outlines

SirSkorpan opened the pull request:
Figured I might as well just create the pull request, I can always squash the commits later.

I've added what I hope is a solution to using Docstrings with Scenario Outlines as described in #7 and #79. The implementation is concentrated to `src/parsed-feature-loading.ts` I also added a couple of examples for TypeScript and ECMAScript to illustrate the new functionality (it was previously not possible to use Docstrings and Scenario Outlines together).

Have a look and ping me if there are any concerns.

bencompton commented on src/parsed-feature-loading.ts:
```diff
@@ -90,24 +90,40 @@ const parseScenarioOutlineExampleSteps = (exampleTableRow: any, scenarioSteps: P
+                    scenarioStep.stepArgument instanceof String
+                ) {
+                    Object.keys(exampleTableRow).forEach((nextTableColumn) => {
+                        stepArgument = (stepArgument as string).replace(
```
It's possible that a doc string might have a field specified more than once, but this code will only replace one instance. It would be better to use a regex with /g to ensure each column gets replaced globally.

SirSkorpan replied:
Ah, nice catch that would indeed be quite possible. I'll update the tests to reflect this as well. 

I believe this is a problem also with the current implementation of scenario outlines and placeholders in steps and arguments (titles use regex an works fine). Do you want me to change those into regex checks as well? (I'll just have to come up with some good test scenario where you'd write something like `Given I <placeholder> <placeholder>` :thinking:

SirSkorpan replied:
The fix now uses regex.

bencompton commented on examples/ecmascript/specs/features/using-docstrings.feature:
```diff
@@ -0,0 +1,21 @@
+Feature: Certificate manufaturing
```
Looks like a typo: manufacturing is missing the "c".

SirSkorpan replied:
Indeed :)

SirSkorpan replied:
fixed

bencompton commented:
@SirSkorpan - thanks for contributing this fix! I commented on a couple of small tweaks, but it looks good otherwise. Once those issues are resolved, I can merge and publish.

I will be publishing this in a beta package for now instead of in the mainline because I've been a bad maintainer the last several months, and master still has some beta functionality in it from a while back (#27).

SirSkorpan commented:
@bencompton I added a few tests for using repeating <placeholder>'s in steps, docstrings and gherkin tables. Hopefully you feel that those belong in this PR as well. As I was adding regex in the replace method for my new code I figured I might as well use it in the old code as well to add support not just to docstrings.

Also added some information to the `docs` about placeholders in scenario outlines as well as docstrings. Have a look an get back to me if you feel something is missing or doesn't belong in this PR.

bencompton commented:
@SirSkorpan - Thanks for fixing the replacement in both places, and also adding additional examples! Your changes are now merged and published to v2.0.13 (tagged as "next").
